File: solutions/go/series/1/series.go
package series

func All(n int, s string) []string {
	result := make([]string, 0, len(s))
	for i := 0; n <= len(s); i++ {
		result = append(result, s[i:n])
		n++
	}
	return result
}

func UnsafeFirst(n int, s string) string {
	return s[:n]
}

func First(n int, s string) (first string, ok bool) {
	if n > len(s) {
		return "", false
	}

	return s[:n], true
}

File: solutions/clojure/series/1/src/series.clj
(ns series
  (:require [clojure.string :as str]))

(defn- slices-helper
  [string length result]
  (if (< (count string) length)
    (vec result)
    (slices-helper
     (str/join "" (rest string))
     length
     (conj result (str/join "" (take length string))))))

(defn slices
  [string length]
  (cond 
    (= length 0) (vector "")
    (or (empty? string) (< (count string) length)) (vector)
    :else (slices-helper string length [])))
